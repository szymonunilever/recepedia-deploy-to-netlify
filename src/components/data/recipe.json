{
  "id": "069cee1a-b673-51ae-829a-cfac76de0223",
  "shortTitle": "Salada Francesa do @caio",
  "recipeDetails": {
    "serves": 2,
    "servingSize": 2,
    "readyTime": 0,
    "totalTime": 35,
    "cookingTime": 25,
    "preparationTime": 10,
    "ovenTime": 0,
    "restTime": 0,
    "marinateTime":  0,
    "freezeTime": 0,
    "chillTime": 0,
    "brewTime": 0
  },
  "description": "",
  "imageUrls": [],
  "nutrients": [],
  "nutrientsPerServing": [],
  "nutrientsPer100g": [],
  "calculatedServingWeightg": 0,
  "creationTime": "2019-05-13T16:06:08Z",
  "title": "Salada Francesa do @caio",
  "ingredients": [
    {
      "productId": null,
      "description": "alface lisa"
    },
    {
      "productId": null,
      "description": "alface roxa"
    },
    {
      "productId": null,
      "description": "tomate"
    },
    {
      "productId": null,
      "description": "cenoura"
    },
    {
      "productId": null,
      "description": "beterraba"
    },
    {
      "productId": null,
      "description": "Queijo comte (mas pode ser o que quiser)"
    },
    {
      "productId": null,
      "description": "pão baguette"
    },
    {
      "productId": null,
      "description": "filé filé mignon"
    }
  ],
  "methods": [
    {
      "position": 1,
      "description": "Assar a carne na frigideira, temperar com manteiga, sal e pimenta do reino"
    },
    {
      "position": 2,
      "description": "Cortar e lavar as alfaces, cenoura, beterraba e tomate"
    },
    {
      "position": 3,
      "description": "Assar a baguette no forno pra comer quentinha"
    },
    {
      "position": 4,
      "description": "Cortar o queijo em tiras"
    }
  ],
  "categories": [
    {
      "name": "difficulties",
      "label": "difficulties",
      "tags":[
      {
        "id": 12231,
        "name": "Easy"
      }
    ]

    }
  ]
}
