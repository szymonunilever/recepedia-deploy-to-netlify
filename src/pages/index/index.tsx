import { graphql } from 'gatsby';
import React from 'react';
import Layout from 'src/components/Layout/Layout';
import SEO from 'src/components/Seo/Seo';
import { Text, TagName } from 'src/components/lib/components/Text';

import { RecipeListing } from 'src/components/lib/components/RecipeListing';
import Hero from 'src/components/lib/components/Hero';
import { findPageComponentContent } from 'src/utils';
import { RecipeItem } from 'src/components/lib/components/RecipeListing/partials';

const HomePage = ({ data }: HomePageProps) => {
  const page = data.allPage.edges[0].node;
  const components = page.components.items;
  const recipes = data.allRecipe.nodes;

  return (
    <Layout>
      <SEO title="Recepedia Home" />

      <Text tag={TagName['h1']} text={data.allPage.edges[0].node.title} />
      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'LatestAndGreatest'
          )}
          list={recipes}
        />
      </section>

      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'TopRecipes'
          )}
          list={recipes}
        />
      </section>

      <section>
        <Hero
          content={findPageComponentContent(components, 'Hero')}
          viewType="Image"
        />
      </section>
    </Layout>
  );
};

export default HomePage;

export const pageQuery = graphql`
  {
    allPage(filter: { type: { eq: "Home" } }) {
      edges {
        node {
          ...PageFields
        }
      }
    }
    allRecipe(skip: 10) {
      nodes {
        ...RecipeFields
      }
    }
  }
`;

interface HomePageProps {
  data: {
    allPage: {
      edges: Edge<PageNode>[];
    };
    allRecipe: {
      nodes: RecipeItem[];
    };
  };
}

interface Edge<T> {
  node: T;
}

interface PageNode {
  components: {
    items: {
      name: string;
      content: any;
    }[];
  };
  title: string;
  type: string;
}
