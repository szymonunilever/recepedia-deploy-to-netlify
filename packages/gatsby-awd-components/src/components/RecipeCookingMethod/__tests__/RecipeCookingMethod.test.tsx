import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { RecipeCookingMethod, RecipeCookingMethodProps } from '../index';
import toJson from 'enzyme-to-json';
import recipe from '../../../mocks/recipe.json';

describe('<RecipeCookingMethod />', () => {
  let wrapper: ReactWrapper;
  const methodList: RMSData.CookingMethodGroup[] = recipe.methods;
  const props: RecipeCookingMethodProps = {
    methodList,
    content: { title: 'Title', subtitle: 'Subtitle' },
  };

  beforeEach(() => {
    wrapper = mount(<RecipeCookingMethod {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("shouldn't render methods", () => {
    wrapper = mount(<RecipeCookingMethod methodList={[]} content={{}} />);
    expect(wrapper.find('.cooking-methods__group')).toHaveLength(0);
  });
});
