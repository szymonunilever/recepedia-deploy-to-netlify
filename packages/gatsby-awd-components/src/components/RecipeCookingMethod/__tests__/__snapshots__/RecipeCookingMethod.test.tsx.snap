// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeCookingMethod /> matches the snapshot 1`] = `
<Component
  content={
    Object {
      "subtitle": "Subtitle",
      "title": "Title",
    }
  }
  methodList={
    Array [
      Object {
        "list": Array [
          Object {
            "description": "Use uma forma pequena de fundo removível de 17,5cm de diâmetro. Molhe um papel manteiga (para ficar maleável) e forre a forma toda, inclusive as laterais.",
            "position": 1,
          },
          Object {
            "description": "Pré-aqueça o forno à 200°C por 10 minutos.",
            "position": 2,
          },
        ],
        "position": 0,
        "title": "desagrupados",
      },
      Object {
        "list": Array [
          Object {
            "description": "Pique as tâmaras e castanhas em pedaços pequenos. Em um processador, bata todos os ingredientes da base. Se necessário, adicione óleo de coco para bater. Retire a mistura do processador, pegue um pouco com as mãos, aperte para ficar mais compacto e vá forrando o fundo da forma, apertando bem com os dedos. A farofinha fica solta mesmo, mas no forno ela dá a liga. Reserve.",
            "position": 3,
          },
        ],
        "position": 1,
        "title": "Base",
      },
      Object {
        "list": Array [
          Object {
            "description": "Leve a polpa de maracujá e o açúcar ao fogo alto. Quando começar a ferver, abaixe o fogo e mexa até engrossar levemente e dissolver o açúcar. Aproximadamente 7 minutos. Leve todos os ingredientes ao processador e bata até ficar homogêneo. 
Leve ao forno por 20 minutos. Deixe esfriar e leve ao congelador por 8 horas.",
            "position": 4,
          },
        ],
        "position": 2,
        "title": "Recheio",
      },
      Object {
        "list": Array [
          Object {
            "description": "Prepare a ganache próximo do momento de servir. Prepare primeiro o creme de castanhas, batendo 1/4 xícara de chá de amêndoa ou castanha de caju com 1/2 xícara de chá de água. Você também pode usar o próprio leite de coco ou creme de leite de soja. Derreta o chocolate no microondas ou banho-maria.",
            "position": 5,
          },
          Object {
            "description": "No micro-ondas, ajuste a potência para \\"média\\". Coloque o chocolate em uma tigela apropriada e aqueça por 1 minuto. Retire, mexa com uma colher para ir homogeneizando e leve novamente por mais 1 minuto ao micro-ondas, repetindo o processo até derreter por completo (2 ou 3 vezes, no máximo). Acrescente o creme de castanhas e misture bem, até virar um creme homogêneo. Você pode acrescentar um pouco de leite de coco para ficar mais fluída, porque a ganache tem textura bem densa.",
            "position": 6,
          },
          Object {
            "description": "Cubra a torta com a ganache e decore como desejar. Eu usei lascas de chocolate 70%, mas você pode colocar gotas de chocolate, nibs de cacau, castanhas picadas ou servir assim mesmo ao que já fica bem bom.",
            "position": 7,
          },
        ],
        "position": 3,
        "title": "Ganache",
      },
    ]
  }
>
  <div
    className="recipeCookingMethod"
    data-componentname="recipeCookingMethod"
    data-componentview="recipeCookingMethod"
  >
    <Text
      className="recipeCookingMethod__title recipe-cooking-method__title"
      tag={1}
      text="Title"
    >
      <h2
        className="recipeCookingMethod__title recipe-cooking-method__title"
      >
        Title
      </h2>
    </Text>
    <Text
      className="recipeCookingMethod__subtitle recipe-cooking-method__subtitle"
      tag={2}
      text="Subtitle"
    >
      <h3
        className="recipeCookingMethod__subtitle recipe-cooking-method__subtitle"
      >
        Subtitle
      </h3>
    </Text>
    <div
      className="cookingMethods cooking-methods"
    >
      <dl
        className="cookingMethods__group cooking-methods__group"
        key="0"
      >
        <dt
          className="cookingMethods__groupTitle cooking-methods__group-title"
        >
          desagrupados
        </dt>
        <dd
          className="cookingMethods__groupItem cooking-methods__group-item"
          key="0"
        >
          Use uma forma pequena de fundo removível de 17,5cm de diâmetro. Molhe um papel manteiga (para ficar maleável) e forre a forma toda, inclusive as laterais.
        </dd>
        <dd
          className="cookingMethods__groupItem cooking-methods__group-item"
          key="1"
        >
          Pré-aqueça o forno à 200°C por 10 minutos.
        </dd>
      </dl>
      <dl
        className="cookingMethods__group cooking-methods__group"
        key="1"
      >
        <dt
          className="cookingMethods__groupTitle cooking-methods__group-title"
        >
          Base
        </dt>
        <dd
          className="cookingMethods__groupItem cooking-methods__group-item"
          key="0"
        >
          Pique as tâmaras e castanhas em pedaços pequenos. Em um processador, bata todos os ingredientes da base. Se necessário, adicione óleo de coco para bater. Retire a mistura do processador, pegue um pouco com as mãos, aperte para ficar mais compacto e vá forrando o fundo da forma, apertando bem com os dedos. A farofinha fica solta mesmo, mas no forno ela dá a liga. Reserve.
        </dd>
      </dl>
      <dl
        className="cookingMethods__group cooking-methods__group"
        key="2"
      >
        <dt
          className="cookingMethods__groupTitle cooking-methods__group-title"
        >
          Recheio
        </dt>
        <dd
          className="cookingMethods__groupItem cooking-methods__group-item"
          key="0"
        >
          Leve a polpa de maracujá e o açúcar ao fogo alto. Quando começar a ferver, abaixe o fogo e mexa até engrossar levemente e dissolver o açúcar. Aproximadamente 7 minutos. Leve todos os ingredientes ao processador e bata até ficar homogêneo. 
Leve ao forno por 20 minutos. Deixe esfriar e leve ao congelador por 8 horas.
        </dd>
      </dl>
      <dl
        className="cookingMethods__group cooking-methods__group"
        key="3"
      >
        <dt
          className="cookingMethods__groupTitle cooking-methods__group-title"
        >
          Ganache
        </dt>
        <dd
          className="cookingMethods__groupItem cooking-methods__group-item"
          key="0"
        >
          Prepare a ganache próximo do momento de servir. Prepare primeiro o creme de castanhas, batendo 1/4 xícara de chá de amêndoa ou castanha de caju com 1/2 xícara de chá de água. Você também pode usar o próprio leite de coco ou creme de leite de soja. Derreta o chocolate no microondas ou banho-maria.
        </dd>
        <dd
          className="cookingMethods__groupItem cooking-methods__group-item"
          key="1"
        >
          No micro-ondas, ajuste a potência para "média". Coloque o chocolate em uma tigela apropriada e aqueça por 1 minuto. Retire, mexa com uma colher para ir homogeneizando e leve novamente por mais 1 minuto ao micro-ondas, repetindo o processo até derreter por completo (2 ou 3 vezes, no máximo). Acrescente o creme de castanhas e misture bem, até virar um creme homogêneo. Você pode acrescentar um pouco de leite de coco para ficar mais fluída, porque a ganache tem textura bem densa.
        </dd>
        <dd
          className="cookingMethods__groupItem cooking-methods__group-item"
          key="2"
        >
          Cubra a torta com a ganache e decore como desejar. Eu usei lascas de chocolate 70%, mas você pode colocar gotas de chocolate, nibs de cacau, castanhas picadas ou servir assim mesmo ao que já fica bem bom.
        </dd>
      </dl>
    </div>
  </div>
</Component>
`;
